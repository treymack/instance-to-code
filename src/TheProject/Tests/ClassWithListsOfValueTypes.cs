using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace TheProject.Tests
{
    [TestFixture]
    public class ClassWithListsOfValueTypes
    {
        [Test]
        public void CanScriptToCode()
        {
            var instance = new ValueListsClass
            {
                Ints = new List<int> { 1, 2, 3 },
                Strings = new List<string> { "one", "two", "three" },
            };

            var code = ITC.Instance.ToCode(instance);

            Approvals.Verify(code);
        }
    }
  
    public class ValueListsClass
    {
        public List<int> Ints { get; set; }
        public List<string> Strings { get; set; }
    }
}
