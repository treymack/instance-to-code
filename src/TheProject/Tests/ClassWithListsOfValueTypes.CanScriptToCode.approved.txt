new TheProject.Tests.ValueListsClass()
{
Array1 = new System.Int32[]
{1, 
2, 
},
Ints = new System.Collections.Generic.List<System.Int32>
{1, 
2, 
3, 
},
NullArray = null, 
Strings = new System.Collections.Generic.List<System.String>
{"one", 
"two", 
"three", 
},
}